src/metadata.rs

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of CycloneDX Rust Cargo.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+use std::io;
+use xml_writer::XmlWriter;
+
+use cargo::core::{Package, TargetKind};
+use chrono::{DateTime, Utc};
+use serde::Serialize;
+use std::str::FromStr;
+
+use crate::author::Author;
+use crate::component::Component;
+use crate::traits::ToXml;
+
+#[derive(Serialize)]
+pub struct Metadata<'a> {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timestamp: Option<DateTime<Utc>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub authors: Option<Vec<Author>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub component: Option<Component<'a>>,
+}
+
+impl<'a> Default for Metadata<'a> {
+    fn default() -> Self {
+        Self {
+            timestamp: Some(Utc::now()),
+            authors: None::<Vec<Author>>,
+            component: None::<Component>,
+        }
+    }
+}
+
+impl<'a> From<&'a Package> for Metadata<'a> {
+    fn from(pkg: &'a Package) -> Self {
+        Self {
+            authors: get_authors_from_package(pkg),
+            component: Some(if could_be_application(pkg) {
+                Component::application(pkg).without_scope()
+            } else {
+                Component::library(pkg).without_scope()
+            }),
+            ..Default::default()
```

Comment:
Does the default `timestamp` of `Some(Utc::now())` make sense for getting metadata from packages?

Reply:
I'm not sure if I follow, mind expanding? The timestamp is more or less when you generated the Bom.

Reply:
Ahhh yeah that was not super clear in retrospect. I was wondering if this function and the one below it related to dependencies as well as top-level packages. If so, does the timestamp of a dependency make sense as `now`, the upload time of the package to [crates.io](https://crates.io), or `None`?

Reply:
That's a great question to which I do not immediately know the answer. I always thought of the metadata on the bom level, not on the component level. @stevespringett you got any thoughts as to what it means?

Reply:
Per the spec, bom->timestamp states:
> The date and time (timestamp) when the document was created.

So this refers to when the SBOM file is created, not when the component was created.